packages/protocol/test-sol/unit/common/GoldToken.t.sol

```diff
@@ -126,6 +127,26 @@ contract GoldTokenTest_transfer is GoldTokenTest {
     vm.expectRevert();
     celoToken.transfer(address(0), ONE_CELOTOKEN);
   }
+
+  function test_Succeeds_whenTransferingToCeloUnreleasedTreasury() public {
+    vm.prank(sender);
+    uint256 balanceBefore = celoToken.balanceOf(celoUnreleasedTreasuryAddress);
+
+    celoToken.transfer(celoUnreleasedTreasuryAddress, ONE_CELOTOKEN);
+    uint256 balanceAfter = celoToken.balanceOf(celoUnreleasedTreasuryAddress);
+    assertGt(balanceAfter, balanceBefore);
```

Comment:
shouldn't this fail?

Reply:
no, that's why we are now using the internal accounting. ERc20 transfers will succeeed, but we do not account for in in the release.

This is done to avoid checking the `to` address on every CELO token transfers.

See [above comment](https://github.com/celo-org/celo-monorepo/pull/11222#discussion_r1772873117) for more details

Reply:
This follows the same pattern as all other core contracts. No fallback implemented but still receive ERC20 transfers. Since it now keeps an internal accounting of the remaining distribution, any transfers are ignored.